src/utils/logger.ts

```diff
@@ -0,0 +1,70 @@
+import { Logger } from '@ar.io/sdk';
+import { captureException, captureMessage } from '@sentry/react';
+
+type LogLevel = 'debug' | 'info' | 'warn' | 'error';
+type LogContext = Record<string, any>;
+
+export class SentryLogger extends Logger {
+  private context: LogContext = {};
+  private component: string;
+  private logLevel: LogLevel;
+
+  constructor({
+    component,
+    initialContext = {},
+    logLevel = 'info',
+  }: {
+    component: string;
+    initialContext?: LogContext;
+    logLevel?: LogLevel;
+  }) {
+    super({ level: logLevel });
+    this.component = component;
+    this.context = initialContext;
+    this.logLevel = logLevel;
+  }
+
+  setContext(context: LogContext): SentryLogger {
+    this.context = { ...this.context, ...context };
+    return this;
+  }
+
+  withContext(context: LogContext): any {
+    return new SentryLogger({
+      component: this.component,
+      initialContext: { ...this.context, ...context },
+      logLevel: this.logLevel,
+    });
+  }
+
+  warn(message: string, context?: LogContext): void {
+    this.warn(message, context);
```

Comment:
Shouldn't this be `super.warn` otherwise you'd have an infinite recursion?!

Reply:
Same goes for `error` function.

Reply:
actually yes - i shouldnt be extending logger here but rather consuming a logger and implemented the logger interface and wrapping the requisite functions